### ISSUE TITLE
Import stops at a file

### ISSUE THREAD

**Issue by nonoumasy** [benyap/firestore-backfire#351]
I tried importing data back to a test Firestore project to test out importing with this library. The import was taking more than an hour so I stopped it. I tried again several times with the same result. How long would a 88mb file take to import? I used this:

```
backfire import hm.ndjson --paths stories -K testCredentials.json -P test-hm-bq0gl2 --verbose true
```

When I look at the logs, it stops at a file.

**Comment by benyap** [comment]
Hi @nonoumasy,

88mb of data definitely shouldn't take over an hour to import 😅 I can see in your screenshot that there were quite a few documents that failed to import. Were there any log messages that described why those imports failed?

I suspect those errors may be blocking the thread.

Could you also please provide the following:

- The entire output log of your import (please attach as a txt file)
- A sample of the different types of data structures you're trying to import, e.g. provide me a document from the stories collection, and a document at the stores/xxx/events collection etc. (please remove any sensitive data)

**Comment by nonoumasy** [comment]
Hi @benyap,

I didn't see any messages that described the failed imports. It just runs and then at some point, it stops...
![Screenshot 2023-05-23 at 7 32 24 PM](https://github.com/benyap/firestore-backfire/assets/6312676/74d1fe96-59c2-4919-b2a2-f0fa6d66d6d0)

Here is a sample of of the data structures used in both stories and events collections:
[temp.ndjson.zip](https://github.com/benyap/firestore-backfire/files/11546552/temp.ndjson.zip)

How would I save the  output log of your import?

**Comment by benyap** [comment]
Hey @nonoumasy, is your Firebase project on the (free) Spark plan? You may be hitting some write limits if you're not on the Blaze plan (pay as you go).

I've been doing some testing on a Firebase project on the Blaze plan. I wrote a script to generate random data, and here is a sample of what I'm uploading - [seed.ndjson.zip](https://github.com/benyap/firestore-backfire/files/11576059/seed.ndjson.zip) - fairly similar to your sample. This is what I was able to achieve with the library:

## Test 1

- Imported 10k documents in under 2 minutes.
- Data size: 24 MB
- Average document size: 2.4 KB

## Test 2

- Imported ~170k documents. Took 32 minutes.
- Data size: 183 MB
- Average document size: 1 KB

## Test 3

- Imported ~430k documents. Took 1h 20m.
- Data size: 832 MB
- Average document size: 1.9 KB

**Comment by nonoumasy** [comment]
@benyap  YES! it is a Spark plan. that makes sense. ok I'll switch it and test. I'll let you know

**Comment by nonoumasy** [comment]
@benyap 
to follow up, I changed to Blaze plan and tried to import again. It seems to be stopping again. If you were able to upload that file in under 2 minutes, then there is still an issue. I keep refreshing during the upload and I let it go for 10 minutes and it wasn't changing.

**Comment by nonoumasy** [comment]
@benyap I tried again today(atleast 12 hours later), it had the same result. It stopped at a file. 
[Terminal Saved Output.txt](https://github.com/benyap/firestore-backfire/files/11581053/Terminal.Saved.Output.txt)

I'll email the data to you.

**Comment by nonoumasy** [comment]
@benyap do you think it has to do with cold start? I don't use that project at all except for tests. What you described also sounds like a 'cold start' issue.

**Comment by benyap** [comment]
Hi @nonoumasy,

I don't think it's a cold start problem. The testing I mentioned in my previous comment was also done on a test project I made just to test Firestore.

Thanks for sending your file through. I've tried importing your data into my test project. I was able to get it to work successfully a few times (all uploaded under 2 minutes). However, I also did encounter the freezing of the program you mentioned several times too. It certainly seems very inconsistent and I don't have an answer as to why yet.

There's a chance it could have something to do with the content in your documents as I notice there's some characters from other languages. I'll dig deeper and I'll let you know if I work anything out 😁

**Comment by nonoumasy** [comment]
> Hi @nonoumasy,
> 
> I don't think it's a cold start problem. The testing I mentioned in my previous comment was also done on a test project I made just to test Firestore.
> 
> Thanks for sending your file through. I've tried importing your data into my test project. I was able to get it to work successfully a few times (all uploaded under 2 minutes). However, I also did encounter the freezing of the program you mentioned several times too. It certainly seems very inconsistent and I don't have an answer as to why yet.
> 
> There's a chance it could have something to do with the content in your documents as I notice there's some characters from other languages. I'll dig deeper and I'll let you know if I work anything out 😁

Thanks @benyap yes. my data has a few properties like description which is translated to storyDescriptionTranslated and eventDescriptionTranslated (depending if its the main story collection or the event collection).  Cheers
